runtime/drive_handler_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"testing"
+
+	"github.com/firecracker-microvm/firecracker-go-sdk"
+	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
+	ops "github.com/firecracker-microvm/firecracker-go-sdk/client/operations"
+	"github.com/firecracker-microvm/firecracker-go-sdk/fctesting"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestStubDriveHandler(t *testing.T) {
+	const tempPath = "test"
+	err := os.Mkdir(tempPath, os.ModePerm)
+	assert.NoError(t, err)
+	defer func() {
+		os.RemoveAll(tempPath)
+	}()
+
+	handler := newStubDriveHandler(tempPath)
+	paths, err := handler.StubDrivePaths(5)
+	assert.NoError(t, err)
+	assert.Equal(t, 5, len(paths))
+
+	infos, err := ioutil.ReadDir(tempPath)
+	assert.NoError(t, err)
+	assert.Equal(t, 5, len(infos))
+}
+
+func TestPatchStubDrive(t *testing.T) {
+	ctx := context.Background()
+	index := 0
+	expectedReplacements := []string{
+		"/correct/path0",
+		"/correct/path1",
+		"/correct/path2",
+	}
+
+	mockClient := &fctesting.MockClient{
+		PatchGuestDriveByIDFn: func(params *ops.PatchGuestDriveByIDParams) (*ops.PatchGuestDriveByIDNoContent, error) {
+			assert.Equal(t, expectedReplacements[index], firecracker.StringValue(params.Body.PathOnHost))
+			index++
+
+			return nil, nil
+		},
+	}
+
+	fcClient := firecracker.NewClient("/path/to/socket", nil, false, firecracker.WithOpsClient(mockClient))
+	client, err := firecracker.NewMachine(ctx, firecracker.Config{}, firecracker.WithClient(fcClient))
+	assert.NoError(t, err, "failed to create new machine")
+
+	handler := stubDriveHandler{
+		drives: []models.Drive{
+			{
+				DriveID:    firecracker.String("stub0"),
+				PathOnHost: firecracker.String("/fake/stub/path0"),
+			},
+			{
+				DriveID:    firecracker.String("stub1"),
+				PathOnHost: firecracker.String("/fake/stub/path1"),
+			},
+			{
+				DriveID:    firecracker.String("stub2"),
+				PathOnHost: firecracker.String("/fake/stub/path2"),
+			},
+		},
+	}
+
+	expectedDriveIDs := []string{
+		"stub0",
+		"stub1",
+		"stub2",
+	}
+	for i, path := range expectedReplacements {
+		driveID, err := handler.PatchStubDrive(ctx, client, path)
+		assert.NoError(t, err, "failed to patch stub drive")
+		assert.Equal(t, expectedDriveIDs[i], firecracker.StringValue(driveID), "drive ids are not equal")
+	}
+}
```

Comment:
Can you add a test for PatchStubDrives that calls PatchStubDrives concurrently?  We should verify that the logic around `atomic.LoadInt64` and `atomic.AddInt64` is correct.